pkg/dogstatsd/server.go

```diff
@@ -108,78 +116,81 @@ func (s *Server) handleMessages(metricOut chan<- *metrics.MetricSample, eventOut
 
 func (s *Server) worker(metricOut chan<- *metrics.MetricSample, eventOut chan<- metrics.Event, serviceCheckOut chan<- metrics.ServiceCheck) {
 	for {
-		s.RLock()
-		if s.Started == false {
-			s.RUnlock()
+		select {
+		case <-s.stopChan:
 			return
-		}
-		s.RUnlock()
-
-		packet := <-s.packetIn
-		var originTags []string
-
-		if packet.Origin != listeners.NoOrigin {
-			var err error
-			log.Tracef("dogstatsd receive from %s: %s", packet.Origin, packet.Contents)
-			originTags, err = tagger.Tag(packet.Origin, false)
-			if err != nil {
-				log.Errorf(err.Error())
-			}
-			log.Tracef("tags for %s: %s", packet.Origin, originTags)
-		} else {
-			log.Tracef("dogstatsd receive: %s", packet.Contents)
-		}
-
-		for {
-			message := nextMessage(&packet.Contents)
-			if message == nil {
-				break
-			}
-
-			if s.Statistics != nil {
-				s.Statistics.StatEvent(1)
-			}
+		case <-s.healthTicker.C:
```

Comment:
you just added the select here, that cause the diff to be huge right? no need to check all the code?

(the logs bellow are not capitalized though 😁 )